File: dev-docs/snippets/document-blob/uploads.ts
import type { JsBaoClient } from "js-bao-wss-client";

export function uploads(client: JsBaoClient, documentId: string) {
  // #region example
  const blobs = client.document(documentId).blobs();
  // Status of all tracked uploads for this document.
  const statuses = blobs.uploads();
  const pending = statuses.filter((u) => u.status === "pending");
  // #endregion example
  return pending;
}

File: dev-docs/snippets/document-blob/uploads.swift
import JsBaoClient

func uploads(client: JsBaoClient, documentId: String) {
  // #region example
  let blobs = client.documents.blobs(documentId: documentId)
  // Status of all tracked uploads for this document (newest first).
  let statuses = blobs.uploads()
  let pending = statuses.filter { $0.status == "pending" }
  // #endregion example
  _ = pending
}
